### PR TITLE
feat(general): fix circleci yaml-doc

### PR DESCRIPTION
**Pull request by maxamel**
This PR fixes case of mismatched type in yaml-doc for resolving sub name in workflow content.
Also a resource test file is added to vcs scanning.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Eliran-Turgeman — APPROVED** [review]

File: tests/github/resources/github_conf/empty_collabs/repository_collaborators.json [hunk 1]
```diff
@@ -0,0 +1 @@
+[]
```
Comment (marynaKK) [inline tests/github/resources/github_conf/empty_collabs/repository_collaborators.json:1]: why change this? :)

**Review by marynaKK — COMMENTED** [review]
We also might want to add a test for it:
```
jobs:
    build_job:
      steps:
        - checkout
```

After debugging this is not the error, but this:
```
jobs:
    test-job-single-line: run/script
```
commenting just for documentation :)

File: tests/github/resources/github_conf/empty_collabs/repository_collaborators.json (hunk 1 above)
Comment (maxamel) [reply tests/github/resources/github_conf/empty_collabs/repository_collaborators.json:1]: We have a test for it, but I guess the resource test file was left out

**Review by Eliran-Turgeman — APPROVED** [review]

File: tests/github/resources/github_conf/empty_collabs/repository_collaborators.json (hunk 1 above)
Comment (marynaKK) [reply tests/github/resources/github_conf/empty_collabs/repository_collaborators.json:1]: it's github, so I will just ask to ensure this is intended here?

**Review by marynaKK — APPROVED** [review]

**Pull request merged**